### PR TITLE
Support adding custom operator in c++ and json format

### PR DESCRIPTION
**Pull request by jlxue**

### CODE REVIEW

**Comment by xysmlx** [comment]
It seems that NNFusion also needs custom frontend converter (e.g., ONNX node to NNFusion node for custom op). Does this PR plan to support it?

**Comment by wenxcs** [comment]
> It seems that NNFusion also needs custom frontend converter (e.g., ONNX node to NNFusion node for custom op). Does this PR plan to support it?

I think we have a fallback policy to treat unrecognized op from frontend

File: src/contrib/custom_op/README.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Adding Custom Operator in NNFusion
+
+## When do you need to add custom operator?
+1. There is an operator that NNFusion doesn't support yet;
```
Comment (xysmlx) [inline src/contrib/custom_op/README.md:4]: Shall we have an example to show this case (i.e., define the custom operator following the automatic frontend converter's format)?

Comment (xysmlx) [reply src/contrib/custom_op/README.md:4]: BTW, shall we add the doc link to the docs/ folder?

**Review by wenxcs — APPROVED** [review]

**Comment by mzmssg** [comment]
> > It seems that NNFusion also needs custom frontend converter (e.g., ONNX node to NNFusion node for custom op). Does this PR plan to support it?
> 
> I think we have a fallback policy to treat unrecognized op from frontend

I checked code, there is no fallback in onnx frontend

File: src/contrib/custom_op/json/custom_test_op.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+    "ops": [
+        {
+            "op": "CustomOpTest",
```
Comment (xysmlx) [inline src/contrib/custom_op/json/custom_test_op.json:4]: Shall we require users to implement the identifier field in the custom_op? We can also implement an identifier generator for generic ops, which could avoids the users to implement the identifier for custom ops.

**Pull request merged**